libraries/config/mechanical_material_data.hpp

```diff
@@ -35,26 +35,97 @@ struct mechanical_material_data<T, 1> final {
 };
 
 template<std::floating_point T>
-struct mechanical_material_data<T, 2> final {
+class mechanical_material_data<T, 2> final {
+    void read_elasticity_parameters(std::array<T, 2>& parameter, const nlohmann::json& config, std::string name, std::array<size_t, 2>& null) {
+        if (const auto& conf = config[name]; conf.is_number()) {
+            parameter.fill(conf.get<T>());
+            null[1] = 1;
+        }
+        else if (conf.is_array() && conf.size() == 2) {
+            material = material_t::ANISOTROPIC;
+            if(!conf[0].is_null() && !conf[1].is_null()) {
+                parameter[0] = conf[0].get<T>();
+                parameter[1] = conf[1].get<T>();
+            } else if (conf[0].is_null() && !conf[1].is_null()) {
+                parameter[0] = conf[1].get<T>();
+                parameter[1] = conf[1].get<T>();
+                null[0] = 1;
+            } else if (!conf[0].is_null() && conf[1].is_null()) {
+                parameter[0] = conf[0].get<T>();
+                parameter[1] = conf[0].get<T>();
+                null[1] = 1;
+            } else {
+                null[0] = 1; null[1] = 1;
+            }
+        } else
+            throw std::domain_error{"The " + name + " should be either a number in the isotropic case, "
+                                    "or an array of size 2 in the anisotropic"};
+    }
+    void calculate_elasticity_parameters(const std::array<size_t, 2>& null_E, const std::array<size_t, 2>& null_nu) {
+        // youngs_modulus[1] * poissons_ratio[0] = Ey * nuxy = Ex * nuyx = youngs_modulus[0] * poissons_ratio[1]
+        if (null_E[0] == 0 && null_E[1] == 1) {
+            youngs_modulus[1] = youngs_modulus[0] * poissons_ratio[1] / poissons_ratio[0];
+            return;
+        }
+        if (null_E[0] == 1 && null_E[1] == 0) {
+            youngs_modulus[0] = youngs_modulus[1] * poissons_ratio[0] / poissons_ratio[1];
+            return;
+        }
+        if (null_nu[0] == 0 && null_nu[1] == 1) {
+            poissons_ratio[1] = youngs_modulus[1] * poissons_ratio[0] / youngs_modulus[0];
+            return;
+        }
+        if (null_nu[0] == 1 && null_nu[1] == 0) {
+            poissons_ratio[0] = youngs_modulus[0] * poissons_ratio[1] / youngs_modulus[1];
+            return;
+        }
+    }  
+    bool check_elasticity_parameters(const std::array<size_t, 2>& null_E, const std::array<size_t, 2>& null_nu) {
+        const size_t null = null_nu[0] + null_nu[1] + null_E[0] + null_E[1];
+        if (null == 1) return false;
+        return true;
+    }
+    bool elasticity_parameters_correctness() {
+        if (poissons_ratio[0] <= 0 || poissons_ratio[0] >= 0.5)
+            return true;
```

Comment:
```suggestion
            return false;
```
Потому что false, как правило, это плохо, а true -- хорошо, соответственно в проверке этой функции надо будет сделать замену на `if(!check_elasticity_parameters())`